mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -118,6 +137,20 @@ class WidgetAdapter(
         fun showBottomSheet(sheet: AbstractWidgetBottomSheet, widget: Widget)
     }
 
+    interface DatePickerPresenter {
+        fun showDatePicker(
+            dateTime: LocalDateTime,
+            widget: Widget
+        ): MaterialDatePicker<Long>
+    }
+
+    interface TimePickerPresenter {
+        fun showTimePicker(
+            dateTime: LocalDateTime,
+            widget: Widget
+        ): MaterialTimePicker
+    }
```

Comment:
The pickers should be shown in a bottom sheet and therefore you can re-use `DetailBottomSheetPresenter` here.

Reply:
I don't know how I have to do that. The MaterialDatePicker and MaterialTimePicker are model components. I don't know how to change that to be part of a bottom sheet.

Reply:
Yes, unfortunately they're a `DialogFragment` instead of a `ViewGroup` (or something like that). One interface is sufficient nevertheless, I think: https://github.com/maniac103/openhab.android/commit/09c5652d3bd45c03ff5d3d9c30b2f16302de5476